pkg/provider/azure_standard.go

```diff
@@ -362,6 +362,22 @@ func (az *Cloud) serviceOwnsRule(service *v1.Service, rule string) bool {
 	return strings.HasPrefix(strings.ToUpper(rule), strings.ToUpper(prefix))
 }
 
+func publicIPOwnsFrontendIP(service *v1.Service, fip *network.FrontendIPConfiguration, pip *network.PublicIPAddress) bool {
+	if pip != nil &&
+		pip.ID != nil &&
```

Comment:
Since `pointer.StringDeref` is used on L373, I don't think this check is needed any more?

Reply:
Ditto, I personally do not intend to change existing code.